orchestra/model/remote_cache.py

```diff
@@ -54,9 +54,22 @@ def get_branches_with_update(component):
 
         map_to_threadpool(get_branches_with_update, clonable_components, parallelism=parallelism)
 
+        self._persist_cache()
+
+    def _persist_cache(self):
         with open(self.cache_path, "w") as f:
             json.dump(self._cached_remote_data, f)
 
+    def _set_entry(self, component_name, branch_name, commit, persist_cache=True):
```

Comment:
After a quick check, no one uses `persist_cache=False`. Drop it?

Reply:
It was meant to be used by `rebuild_cache.get_branches_with_update` for updating the cache without immediately writing it to disk, I forgot to update that method. Good catch!

Reply:
I'm not a fan of `bool` arguments, consider splitting in two functions with different names. One calling the other.
No biggie though.

Reply:
Closing after IRL conversation